File: src/router/router.cpp
#include "router.h"
#include "handlers/handlers.h"
#include "../net/server/server.h"

namespace router {
    void init_router(std::string ip,uint_least16_t port){
        net::server::Server server({boost::asio::ip::make_address(ip),port});
        server.append_handler("/auth/login",net::request::Methods::GET,handler::auth::HandlerLogin);
        server.append_handler("/auth/register",net::request::Methods::POST,handler::auth::HandlerRegistration);
        server.run();
    }
}

File: unit-test/src/router/router.go
package router

import "net/http"

type HandlerFuncStruct struct {
	Name     string
	Function func(w http.ResponseWriter, r *http.Request)
}

type Server struct {
	Ip       string
	Port     string
	handlers []HandlerFuncStruct
}

func (s *Server) Run_server() {
	for i := 0; i < len(s.handlers); i++ {
		http.HandleFunc(s.handlers[i].Name, s.handlers[i].Function)
	}
	http.ListenAndServe((s.Ip + ":" + s.Port), nil)
}

func (s *Server) Append_handler(handler HandlerFuncStruct) {
	s.handlers = append(s.handlers, handler)
}
